vpr/src/noc/west_first_routing.cpp

```diff
@@ -23,105 +24,81 @@ const std::vector<TurnModelRouting::Direction>& WestFirstRouting::get_legal_dire
-     * the current router. Therefore, we have two
-     * directions to choose from. A hash value is generated
-     * based on the source, current, and destination router IDs
-     * along with the traffic flow ID. Then, we select one direction
-     * by flipping a biased coin. For example if
-     * 1) dst_router_pos.x - curr_router_pos.x = 2
-     * 2) dst_router_pos.y - curr_router_pos.y = 8
-     * We take the UP direction with 80% chance, and RIGHT with 20%.
-     */
+        if (returned_legal_direction.empty()) {
+            if (dst_router_pos.x > curr_router_pos.x) {
+                returned_legal_direction.push_back(TurnModelRouting::Direction::EAST);
+            }
 
-    // compute the hash value
-    uint32_t hash_val = get_hash_value(src_router_id, dst_router_id, curr_router_id, traffic_flow_id);
-    // get the maximum value that can be represented by size_t
-    const uint32_t max_uint32_t_val = std::numeric_limits<uint32_t>::max();
+            if (dst_router_pos.y > curr_router_pos.y) {
+                returned_legal_direction.push_back(TurnModelRouting::Direction::NORTH);
+            }
 
-    // get the distance from the current router to the destination in each coordination
-    int delta_x = abs(dst_router_pos.x - curr_router_pos.x);
-    int delta_y = abs(dst_router_pos.y - curr_router_pos.y);
-
-    // compute the probability of going to the right direction
-    uint32_t east_probability = delta_x * (max_uint32_t_val / (delta_x + delta_y));
-
-    TurnModelRouting::Direction selected_direction = TurnModelRouting::Direction::INVALID;
-
-    if (hash_val < east_probability) { // sometimes turn right
-        selected_direction = TurnModelRouting::Direction::RIGHT;
-    } else { // if turning right was rejected, take the other option (north or south)
-        selected_direction = select_direction_other_than(legal_directions, TurnModelRouting::Direction::RIGHT);
+            if (dst_router_pos.layer_num > curr_router_pos.layer_num) {
+                returned_legal_direction.push_back(TurnModelRouting::Direction::UP);
+            } else if (dst_router_pos.layer_num < curr_router_pos.layer_num) {
+                returned_legal_direction.push_back(TurnModelRouting::Direction::DOWN);
+            }
+        }
+    } else {    // 2D NoC
+        if (dst_router_pos.x < curr_router_pos.x) {
+            returned_legal_direction.push_back(TurnModelRouting::Direction::WEST);
+        } else { // to the east or the same column
+            if (dst_router_pos.x > curr_router_pos.x) { // not the same column
+                returned_legal_direction.push_back(TurnModelRouting::Direction::EAST);
+            }
+
+            if (dst_router_pos.y > curr_router_pos.y) {
+                returned_legal_direction.push_back(TurnModelRouting::Direction::UP);
+            } else if (dst_router_pos.y < curr_router_pos.y) {
+                returned_legal_direction.push_back(TurnModelRouting::Direction::DOWN);
```

Comment:
SOUTH

(Probably should check all UP / DOWN occurences and make sure they should not be NORTH/SOUTH)